### DISCUSSION TITLE
How to handle pynvim in virtual environments

### DISCUSSION THREAD

**Discussion by RunnersNum40** [LazyVim/LazyVim#3135] [Q&A] [answered]
I'm encountering an issue where I need to install pynvim in every Python virtual environment to work seamlessly with LazyVim. Here’s a breakdown of my workflow and what I’m aiming to achieve:

1. Launching from Active Virtual Environment: When I start nvim with a virtual environment active, the LSP automatically uses it, and it works well. Alternatively, I can start nvim without an environment and manually select one using leader-c-v, but I must do this each time.

2. Using System Python in nvim: I have pynvim installed globally, so nvim functions correctly when no virtual environment is active. However, if I start nvim within a virtual environment, I must install pynvim in that environment.

What I Need:

1. I want the LSP in nvim to automatically use the Python environment from which nvim was launched.
2. I'd like nvim to default to using the system Python when not launched from a specific virtual environment.

Is there a way to configure LazyVim to meet these requirements without installing pynvim in each virtual environment?

**Comment by geoffreyvanwyk** [comment, marked as answer]
From `:help provider-python`:

```
PYTHON VIRTUALENVS ~
						*python-virtualenv*
If you plan to use per-project virtualenvs often, you should assign one
virtualenv for Nvim and hard-code the interpreter path via
|g:python3_host_prog| so that the "pynvim" package is not required
for each virtualenv.

Example using pyenv: >bash
    pyenv install 3.4.4
    pyenv virtualenv 3.4.4 py3nvim
    pyenv activate py3nvim
    python3 -m pip install pynvim
    pyenv which python  # Note the path
The last command reports the interpreter path, add it to your init.vim: >vim
    let g:python3_host_prog = '/path/to/py3nvim/bin/python'
```

**Reply by RunnersNum40** [reply]
Thank you this is exactly what I was looking for. I ended up adding these lines to my `lua/config/options.lua` file.

```lua
vim.g.python3_host_prog = "/usr/bin/python"
vim.g.python_host_prog = "/usr/bin/python"
```
